src/solver/softmax/backward_softmaxv3.cpp

```diff
@@ -0,0 +1,186 @@
+    auto isReduceSizeSmall = problem.IsReduceSizeSmall();
+
+    auto kernel        = KernelInfo{};
+    kernel.kernel_file = "MIOpenSoftmaxV3.cpp";
+
+    size_t xlocalsize = 1;
+    size_t xgridsize  = 1;
+    size_t ylocalsize = 1;
+    size_t ygridsize  = 1;
+    size_t zlocalsize = 1;
+    size_t zgridsize  = 1;
+
+    if(isReduceSizeSmall)
+    {
+        kernel.kernel_name = "SoftmaxBwdSmallContiguous";
+        xlocalsize         = LOCAL_SIZE;
+        xgridsize          = outer_size * inner_size * xlocalsize;
+    }
+    else
+    {
+        if(isAllStrideOne)
+        {
+            kernel.kernel_name = "SoftmaxBwdStrideOneContiguous";
+            xlocalsize         = LOCAL_SIZE;
+            xgridsize          = outer_size * inner_size * xlocalsize;
+        }
+        else
+        {
+            kernel.kernel_name = "SoftmaxBwdDimIsNotLastContiguous";
+            xlocalsize         = LOCAL_SIZE;
+            xgridsize = outer_size * ((inner_size + CHUNK_SIZE - 1) / CHUNK_SIZE) * LOCAL_SIZE;
+        }
+    }
+
+    const auto build_params =
+        KernelBuildParameters{{"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+                              {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+                              {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+                              {"IO_TYPE", io_dtype == "bfloat16" ? "ushort" : io_dtype}};
+
+    kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+    kernel.l_wk.push_back(xlocalsize);
+    kernel.l_wk.push_back(ylocalsize);
+    kernel.l_wk.push_back(zlocalsize);
+    kernel.g_wk.push_back(xgridsize);
+    kernel.g_wk.push_back(ygridsize);
+    kernel.g_wk.push_back(zgridsize);
+
+    result.construction_params.push_back(kernel);
+
+    result.invoker_factory =
+        [isAllStrideOne, isReduceSizeSmall, reduce_size, inner_size, outer_size](
+            const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+                decltype(auto) kernel = handle_.Run(kernels.front());
+                decltype(auto) params = raw_params.CastTo<miopen::softmax::InvokeParams>();
+
+                if(isReduceSizeSmall)
+                {
+                    kernel(params.backward_y,
+                           params.dy,
+                           params.dx,
+                           reduce_size,
+                           inner_size,
+                           outer_size,
+                           static_cast<uint32_t>(params.algorithm));
+                }
+                else
+                {
+                    if(isAllStrideOne)
+                    {
+                        kernel(params.backward_y,
+                               params.dy,
+                               params.dx,
+                               reduce_size,
+                               static_cast<uint32_t>(params.algorithm));
```

Comment:
consider to use `int32_t` instead of the unsigned one, because enum can be negative (even though, in current implementation, we can use unsigned types, but i highly recommend to use `int32_t` for general cases).
```suggestion
                               static_cast<int32_t>(params.algorithm));
```